core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -655,7 +660,7 @@ protected void parse() throws ProtocolException {
         lockTime = readUint32();
         optimalEncodingMessageSize += 4;
 
-        if(getVersionShort() >= 3 && getType() != Type.TRANSACTION_NORMAL) {
+        if(getVersionShort() >= MAX_STANDARD_VERSION && getType() != Type.TRANSACTION_NORMAL) {
```

Comment:
This should be DIP2_VERSION, since MAX_STANDARD_VERSION may be raised to 4 at a later time, but this if block is only execute when reading version 3 transactions.